code/modules/supply/supply_pack.dm

```diff
@@ -32,3 +32,29 @@
 		//Add the name to the UI manifest
 		ui_manifest += "[initial(AM.name)]"
 	manifest += "</ul>"
+
+/datum/supply_packs/proc/create_package(atom/spawn_location)
+	var/obj/structure/closet/crate/crate = new containertype(spawn_location)
+
+	crate.name = containername
+	if(access)
+		crate.req_access = list(text2num(access))
+
+	//we now create the actual contents
+	for(var/typepath in generate_items())
+		if(!typepath)
+			continue
+		var/atom/A = new typepath(crate)
+		if(amount && A.vars.Find("amount") && A:amount) // This is sketchy as hell
+			A:amount = amount
+
```

Comment:
```suggestion
		var/atom/A = new typepath(crate)
		if(isstack(A))
				var/obj/item/stack/mats = A
				mats.amount = amount
```
this block of code is only for stacks (I checked)
This could really just do with getting rid of the `var/amount` completely and instead using `/30` or `/50` typepaths for stack crates but whatever.